### PR TITLE
APPS-2685 (Nice-to-have) Print NF workdir at beginning of job log

### PR DESCRIPTION
**Pull request by r-i-v-a**
- [x] Merge other branch this branch was based on first

Testing details on Jira --> APPS-2685

### CODE REVIEW

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh [hunk 1]
```diff
@@ -454,14 +454,15 @@ get_nextflow_environment() {
...
+  echo "=== NF command          :" "${NEXTFLOW_CMD[@]}"
+  echo "=== NF session ID       : ${NXF_UUID}"
+  echo "=== NF workdir          : ${NXF_WORK}"
   if [[ $preserve_cache == true ]]; then
-    echo "=== NF cache folder : dx://${DX_CACHEDIR}/${NXF_UUID}/"
+    echo "=== Preserving cache    : true"
   fi
-  echo "=== NF command      :" "${NEXTFLOW_CMD[@]}"
-  echo "=== Built with dxpy : @@DXPY_BUILD_VERSION@@"
+  echo "=== NF source on worker : @@RESOURCES_SUBPATH@@"
```
Comment (r-i-v-a) [inline src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:463]: Edited wording and order

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh [hunk 2]
```diff
@@ -454,14 +454,15 @@ get_nextflow_environment() {
...
 log_context_info() {
   echo "============================================================="
-  echo "=== NF projectDir   : @@RESOURCES_SUBPATH@@"
-  echo "=== NF session ID   : ${NXF_UUID}"
-  echo "=== NF log file     : dx://${DX_JOB_OUTDIR%/}/${LOG_NAME}"
+  echo "=== NF command          :" "${NEXTFLOW_CMD[@]}"
+  echo "=== NF session ID       : ${NXF_UUID}"
+  echo "=== NF workdir          : ${NXF_WORK}"
   if [[ $preserve_cache == true ]]; then
-    echo "=== NF cache folder : dx://${DX_CACHEDIR}/${NXF_UUID}/"
```
Comment (r-i-v-a) [inline src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:461]: This is now covered by printing "NF workdir..." always

Comment (yihchii) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:461]: Would this be a breaking change to current users? For example, what if users rely on reading/parsing the existing log format? 
What is the down side of keeping the existing log but adding new ones?

Comment (yihchii) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:461]: Please address the command. Once it is resolved, this LGTM. Thanks!

**Review by yihchii — APPROVED** [review]
Please review the review (I am concerned about if i'd be breaking changes) before merge.

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh (hunk 2 above)
Comment (r-i-v-a) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:461]: I reverted the other changes besides adding the "=== NF workdir      : ${NXF_WORK}" line.

**Comment by r-i-v-a** [comment]
Please review, @mhrvol @YuxinShi0423 (review needed from someone with write access).

**Review by YuxinShi0423 — APPROVED** [review]
Looks good to me :)

**Pull request merged**